### PR TITLE
Protocol handler broker t6 & t7

### PR DESCRIPTION
**Pull request by maleck13**
## What does this PR do?
task 6 & 7

- edge case bug fix for meta data only changes
- wiring subscriptions/listen notifications for 2026 upstreams
- e2e tests to prove notifications work for 2026 clients

related #1316 

## Summary by CodeRabbit

- **New Features**
  - Added automatic notifications when available tools change.
  - Tool listings now refresh when caching settings or access scope change.
  - Improved private-tool filtering based on authentication and ownership.
  - Improved reconnection and notification handling for supported connections.

- **Bug Fixes**
  - Tool updates now propagate without requiring a manual refresh.

- **Documentation**
  - Updated security, architecture, caching, notification, and test coverage documentation.

- **Tests**
  - Added end-to-end coverage for dynamic tool updates and client notifications.
  - Increased default parallel test processes from one to two.

### CODE REVIEW

File: internal/tests/stateless-server/server.go [hunk 1]
```diff
@@ -222,19 +238,43 @@ func userToolFilterMiddleware() mcp.Middleware {
...
+					}
+				}
+				filtered := make([]*mcp.Tool, 0, len(toolsResult.Tools))
+				for _, tool := range toolsResult.Tools {
+					if !hidden[tool.Name] {
+						filtered = append(filtered, tool)
+					}
 				}
+				toolsResult.Tools = filtered
 			}
```
[the remark this answers is not in the document]
Comment (maleck13) [reply internal/tests/stateless-server/server.go:288]: this is just a test server the tests ensure to call with the correct or no authorization

**Review by david-martin — DISMISSED** [review]
There's a few ways that tools are added and removed in the test servers. This could be cleaned up in a follow up.

**Review by david-martin — APPROVED** [review]
Re-approving
